test/extended/networking/egress_router_cni.go

```diff
@@ -0,0 +1,96 @@
+	IPv4MatchPattern = "IP Destinations: [80 UDP 10.100.3.0 8080 SCTP 203.0.113.26 80 8443 TCP 203.0.113.27 443]"
+	IPv6MatchPattern = "IP Destinations: [80 UDP 10:100:3::0 8080 SCTP 203:0:113::26 80 8443 TCP 203:0:113::27 443]"
+	TimeOut          = 1 * time.Minute
+	Interval         = 5 * time.Second
+)
+
+var _ = g.Describe("[sig-network][Feature:EgressRouterCNI]", func() {
+	var oc *exutil.CLI
+	oc = exutil.NewCLI(EgressRouterCNIE2E)
+
+	g.It("Should ensure ipv4 egressrouter cni resources are created", func() {
+		doEgressRouterCNI(EgressRouterCNIV4Manifest, oc, IPv4MatchPattern)
+	})
+	InOVNKubernetesContext(
+		func() {
+			g.It("Should ensure ipv6 egressrouter cni resources are created", func() {
+				doEgressRouterCNI(EgressRouterCNIV6Manifest, oc, IPv6MatchPattern)
+			})
+		},
+	)
+})
+
+func doEgressRouterCNI(manifest string, oc *exutil.CLI, matchString string) error {
+	var deployment *appsv1.Deployment
+	var podList *v1.PodList
+	f := oc.KubeFramework()
+
+	g.By("Creating an egressroutercni object")
+	egressRouterCNIYaml := exutil.FixturePath("testdata", "egress-router-cni", manifest)
+
+	g.By(fmt.Sprintf("Calling oc create -f %s", egressRouterCNIYaml))
+	err := oc.AsAdmin().Run("create").Args("-f", egressRouterCNIYaml).Execute()
+	e2e.ExpectNoError(err, "While creating egress-router-cni object")
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	g.By("Making sure egressroutercni deployment is created")
+	o.Eventually(func() error {
+		deployment, err = f.ClientSet.AppsV1().Deployments(f.Namespace.Name).Get(context.Background(), EgressRouterCNIDeployment, metav1.GetOptions{})
+		return err
+	}, TimeOut, Interval).Should(o.Succeed())
+
+	g.By("Checking network-attachment-definition resource is created")
+	output, err := oc.AsAdmin().Run("get").Args("network-attachment-definition", "-o=jsonpath={.items[0].metadata.name}").Output()
+	e2e.ExpectNoError(err, "While creating egress-router-cni network-attachement-definition")
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(output).Should(o.ContainSubstring(EgressRouterCNINadObject))
+
+	g.By("Getting a pod from deployment in running state")
+	o.Eventually(func() error {
+		podList, err = e2edeployment.GetPodsForDeployment(f.ClientSet, deployment)
+		e2e.ExpectNoError(err, "While getting pods from egress-router-cni deployment")
+		return err
+	}, TimeOut, Interval).Should(o.Succeed())
+
+	o.Expect(podList.Items).NotTo(o.BeEmpty())
+	pod := podList.Items[0]
+	e2e.Logf("egress router cni pod %s is created\n", pod.Spec.Containers[0].Name)
+
+	node := pod.Spec.NodeName
+	expectNoError(err)
+	g.By("Checking for specific output in the egress-router log on the node " + node)
+	result, err := oc.AsAdmin().Run("debug").Args("node/"+node, "--", "chroot", "/host", "/bin/bash", "-c", "cat /tmp/egress-router-log").Output()
+	if err != nil {
+		e2e.Failf("couldn't cat egress-router cni log: %v", err)
+	}
+	if !strings.Contains(result, matchString) {
```

Comment:
Are you planning to add more tcs? Checking the logs to verify it has started is better than nothing but why not go the full way. (I'm not egress router expert)

Have you thought about this tc:
1. Launch server which will be egress router destination
2. Launch egress router cni pod that has its destination set to server ip
3. Client sends request to egress router
4. Verify request is received by server

I understand this is not possible directly on cloud environment but someone smarter than me here will surely know how we can get close to this flow.

Reply:
Youre right - layer 2 manipulations are usually not allowed on the cloud AFAIK.

Forgive my ignorance, Is there a baremetal test we can leverage and only activate this test on that?

Reply:
I don't think this can be done in this repo AFAIK there is another repo for BM CI only https://github.com/openshift-metal3/dev-scripts, I think in this repo this is as far as we can go :)